osxkeychain.go

```diff
@@ -137,37 +179,50 @@ func authenticationTypeToGo(authtype C.CFTypeRef) AuthenticationType {
 
 // Adds an Internet password to the user's default keychain.
 func AddInternetPassword(pass *InternetPassword) error {
+	if err := pass.CheckValidity(); err != nil {
+		return err
+	}
+
+	serverName := C.CString(pass.ServerName)
+	defer C.free(unsafe.Pointer(serverName))
+
+	var securityDomain *C.char
+	if len(pass.SecurityDomain) > 0 {
```

Comment:
the idiomatic Go way of doing this for a string is `if pass.SecurityDomain != "" {`. Can you change to that? Otherwise this change looks good